src/lib.rs

```diff
@@ -317,10 +329,142 @@ impl<L, R> Iterator for Either<L, R>
     }
 
     fn fold<Acc, G>(self, init: Acc, f: G) -> Acc
-        where G: FnMut(Acc, Self::Item) -> Acc,
+        where G: FnMut(Acc, L::Item) -> Acc,
```

Comment:
For documentation's sake, I'd use `Self::Item` all the way through. Using `L::Item` is shorter, but could be confusing.